### PR TITLE
feat(test): moving unit testing over to vitest

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

[Vitest](https://vitest.dev/) is fast as hell and drops our unit test execution time down by a crazy amount.

#### Before
```
Test Suites: 21 passed, 21 total
Tests:       5 todo, 620 passed, 625 total
Snapshots:   27 passed, 27 total
Time:        2.836 s, estimated 4 s
Ran all test suites.
npx jest  22.41s user 1.97s system 737% cpu 3.306 total
```

#### Now
```
 Test Files  20 passed (20)
      Tests  619 passed | 5 todo (624)
   Start at  13:24:19
   Duration  1.33s (transform 1.70s, setup 1ms, collect 4.26s, tests 2.05s, environment 2ms, prepare 1.51s)

npx vitest run  6.87s user 1.66s system 485% cpu 1.755 total
```

### CODE REVIEW

File: __tests__/.eslintrc [hunk 1]
```diff
@@ -1,6 +1,13 @@
 {
-  "extends": "@readme/eslint-config/testing",
+  "extends": [
+    "plugin:vitest/all"
```
Comment (erunion) [inline __tests__/.eslintrc:3]: Will replace this with `@readme/eslint-config/testing/vitest` when https://github.com/readmeio/standards/pull/723 is merged and published.

File: __tests__/analyzer/index.test.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import type { OASDocument } from '../../src/rmoas.types';
 
+import { describe, beforeAll, it, expect } from 'vitest';
```
Comment (erunion) [inline __tests__/analyzer/index.test.ts:3]: Vitest offers a globals option[^1] for automatically adding these methods to every unit test but I kind of prefer the explicitness of **not** doing that.

[^1]: https://vitest.dev/config/#globals

**Review by llimllib — APPROVED** [review]
no config! lovely

File: package.json [hunk 3]
```diff
@@ -32,16 +32,17 @@
...
     "prebuild": "rm -rf dist/",
     "prepack": "npm run build",
     "prepare": "husky install",
     "pretest": "npm run lint",
     "prettier": "prettier --list-different \"./**/**.{js,ts,md}\"",
-    "prettier:write": "prettier --list-different --write \"./**/**.{js,ts,md}\"",
-    "test": "tsc; jest --coverage",
-    "test-watch": "tsc; jest --watch",
+    "prettier:fix": "prettier --list-different --write \"./**/**.{js,ts,md}\"",
+    "test": "vitest run --coverage",
```
Comment (kanadgupta) [inline package.json:45]: see [this comment](https://github.com/readmeio/api/pull/691#discussion_r1290700363)

```suggestion
    "test": "vitest --coverage",
```

File: package.json [hunk 4]
```diff
@@ -32,16 +32,17 @@
...
-    "lint": "eslint . --ext .js,.ts",
+    "lint": "npm run lint:types; npm run lint:js; npm run lint:docs",
+    "lint:js": "eslint . --ext .js,.ts",
     "lint:docs": "npx -y alex .",
+    "lint:types": "tsc --noEmit",
     "prebuild": "rm -rf dist/",
     "prepack": "npm run build",
     "prepare": "husky install",
     "pretest": "npm run lint",
     "prettier": "prettier --list-different \"./**/**.{js,ts,md}\"",
```
Comment (kanadgupta) [inline package.json:43]: doesn't have to happen in this PR but thoughts on running prettier against everything?

**Review by kanadgupta — APPROVED** [review]
couple minor comments below but LGTM overall! absolutely obsessed with the fact that there's no config file

File: package.json (hunk 4 above)
Comment (erunion) [reply package.json:43]: i think this is already doing that? we don't have any yaml files in here for testing flows

Comment (kanadgupta) [reply package.json:43]: looks like prettier is flagging a few other files:

CleanShot 2023-08-14 at 11 43 58@2x

**Pull request merged**